### PR TITLE
Add `integrate` GCF function for a serverless log on GCP.

### PR DESCRIPTION
**Pull request by jiggoha**

### CODE REVIEW

File: gcp-log/internal/storage/storage.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+	for {
+		// Pass an empty rootDir since we don't need this concept in GCS.
+		sp := filepath.Join(layout.SeqPath("", end))
+
+		// Read the object in an anonymous function so that the reader gets closed
+		// in each iteration of the outside for loop.
+		numSequenced, err := func() (uint64, error) {
+			r, err := bkt.Object(sp).NewReader(ctx)
+			if err != nil {
+					return end - begin, fmt.Errorf("failed to create reader for object '%s' in bucket '%s': %v", sp, c.bucket, err)
```
Comment (AlCutter) [inline gcp-log/internal/storage/storage.go:145]: nit: can use `%q` for `'%s'`

File: serverless/cmd/gcp/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+TODO(jayhou): make this info more friendly. Currently contains rough notes of
```
Comment (AlCutter) [inline serverless/cmd/gcp/README.md:1]: Ah, it looks like the files are in both locations now in the PR!
I'll stop reviewing here for now :)

Comment (jiggoha) [reply serverless/cmd/gcp/README.md:1]: I have removed the "internal" version now.

File: gcp-log/internal/storage/storage.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	bucket string
+	// nextSeq is a hint to the Sequence func as to what the next available
+	// sequence number is to help performance.
+	// Note that nextSeq may be <= than the actual next available number, but
+	// never greater.
+	nextSeq uint64
+}
+
+// NewClient returns a Client which allows interaction with the log implemented on GCS.
+func NewClient(ctx context.Context, projectID string) (*Client, error) {
```
[the remark this answers is not in the document]
Comment (jiggoha) [reply gcp-log/internal/storage/storage.go:55]: I think we will need this in Sequence. Is it okay with you if I add that when I get there?

[the remark this answers is not in the document]
Comment (AlCutter) [reply gcp-log/internal/storage/storage.go]: Ah sure!

File: serverless/integration/integration_test.go [hunk 4]
```diff
@@ -46,7 +46,7 @@ func RunIntegration(t *testing.T, s log.Storage, f client.Fetcher, lh *rfc6962.H
 	ctx := context.Background()
 	lv := merkle.NewLogVerifier(lh)
 
-	// Do a few interations around the sequence/integrate loop;
+	// Do a few interactions around the sequence/integrate loop;
```
Comment (AlCutter) [inline serverless/integration/integration_test.go:49]: Thanks for correcting my typo - I think it was intended to be "iterations", but "interactions" works too :)

File: serverless/internal/storage/fs/fs.go [hunk 5]
```diff
@@ -107,7 +107,7 @@ func Create(rootDir string) (*Storage, error) {
 // Returns the sequence number assigned to this leaf (if the leaf has already
 // been sequenced it will return the original sequence number and
 // storage.ErrDupeLeaf).
-func (fs *Storage) Sequence(leafhash []byte, leaf []byte) (uint64, error) {
+func (fs *Storage) Sequence(ctx context.Context, leafhash []byte, leaf []byte) (uint64, error) {
```
Comment (AlCutter) [inline serverless/internal/storage/fs/fs.go:110]: I think any of these `ctx` params that are not used in their respective function bodies can be replaced `_`?

**Review by AlCutter — APPROVED** [review]
LGTM, just a small suggestion inline which you can take or leave as you see fit.

File: serverless/integration/integration_test.go (hunk 4 above)
Comment (jiggoha) [reply serverless/integration/integration_test.go]: Oh, I just mindlessly went with lint's suggestion. Switched to "iterations"!

**Pull request merged**